src/components/ImageInput/ImageInput.js

```diff
@@ -1,66 +1,73 @@
 import React, { Component } from 'react';
+import closeIcon from '../../app/img/close-icon.png';
+import { REMOVE_BUTTON_SIZE } from '../../app/utilities/index';
 import './style.css'
 
-class ImageDropzone extends Component {
-  constructor() {
-    super();
+class ImageInput extends Component {
+  constructor(props) {
+    super(props);
     this.state = {
       error: null,
+      images: props.images ? props.images : []
     }
   }
 
-  handleChange = (e) => {
-    const image = e.target.files[0];
-    // this.removeThumbnail(e.target);
-    if (!image.type.startsWith("image/")){
-      this.setState({error: "File must be a picture!"});
-    } else {
-      this.setState({error: null});
-      this.updateThumbnail(e.target, image);
-    }
-
-    const reader = new FileReader();
-    reader.readAsDataURL(image);
-    reader.onload = () => {
-      this.props.onChange(reader.result);
+  componentDidUpdate(prevProps, prevState) {
+    if(prevState.images.length !== this.state.images.length){
+      this.props.onChange(this.state.images);
     }
   }
 
-  updateThumbnail = (inputElement, image) => {
-    const parentElement = inputElement.parentNode;
-    parentElement.querySelector('.dropzone-before').style.display = 'none';
-
-    const thumbnailElement = document.createElement('div');
-    thumbnailElement.classList.add('dropzone-thumbnail');
+  handleChange = (e) => {
+    const images = [...e.target.files];
+    images.forEach(image => {
+      if (!image.type.startsWith("image/")) {
+        this.setState({ error: "File must be a picture!" });
+      } else {
+        this.setState({ error: null });
+      }
 
-    const reader = new FileReader();
-    reader.readAsDataURL(image);
-    reader.onload = () => {
-      thumbnailElement.style.backgroundImage = `url(${ reader.result })`;
-    }
-    parentElement.appendChild(thumbnailElement);
+      const reader = new FileReader();
+      reader.readAsDataURL(image);
+      reader.onload = () => {
+        this.setState({
+          images: [...this.state.images, reader.result]
+        })
+      }
+    })
   }
 
-  removeThumbnail = (inputElement) => {
-    const parentElement = inputElement.parentNode;
-    parentElement.querySelector('.dropzone-before').style.display = 'flex';
-
-    const thumbnail = parentElement.querySelector('.dropzone-thumbnail');
-    if(thumbnail){
-      thumbnail.remove();
-    }
+  handleDeleteImage = (index) => {
+    const prevImages = [...this.state.images];
+    prevImages.splice(index, 1);
+    this.setState({ images: [...prevImages] });
```

Comment:
We could use setState with an optional callback function to make this one clean and we can check the previous state easier. Also, please avoid `splice` which can affect directly the array